src/pam_oauth2_device.cpp

```diff
@@ -335,21 +335,30 @@ bool is_authorized(Config *config,
             return true;
         }
     }
-
     // Try to authorize against LDAP
-    if (!config->ldap_host.empty())
+    if (!config->ldap_hosts.empty())
     {
         size_t filter_length = config->ldap_filter.length() + strlen(username_remote) + 1;
         char *filter = new char[filter_length];
         snprintf(filter, filter_length, config->ldap_filter.c_str(), username_remote);
-        int rc = ldap_check_attr(config->ldap_host.c_str(), config->ldap_basedn.c_str(),
-                                 config->ldap_user.c_str(), config->ldap_passwd.c_str(),
-                                 filter, config->ldap_attr.c_str(), username_local);
+        for (auto ldap_host : config->ldap_hosts)
+        {
+            int rc = ldap_check_attr(ldap_host.c_str(), config->ldap_basedn.c_str(),
+                                     config->ldap_user.c_str(), config->ldap_passwd.c_str(),
+                                     filter, config->ldap_attr.c_str(), username_local);
+            if (rc == LDAPQUERY_TRUE)
+            {
+                delete[] filter;
+                return true;
+            }
+            if (rc == LDAPQUERY_FALSE)
+            {
+                delete[] filter;
+                return false;
+            }
```

Comment:
```suggestion
```

I think this block should be removed. For cases other than `LDAPQUERY_TRUE`, i.e. `LDAPQUERY_FALSE` and `LDAPQUERY_ERROR` the loop should try another LDAP server instead of returning.